lib/patchelf/patcher.rb

```diff
@@ -20,12 +20,25 @@ class Patcher
     #   Filename of input ELF.
     # @param [Boolean] logging
     #   Whether to use a (stderr-initialized) logger for errors
-    def initialize(filename, logging: true)
+    # @param [:log, :silent, :exception] on_error
+    #   :log = logs to stderr
```

Comment:
Add statements when will this option be used
something like "used to decide the action when the desired segment/tag field isn't present"